server/src/main/java/com/saralein/server/logger/Logger.java

```diff
@@ -1,5 +1,9 @@
 package com.saralein.server.logger;
 
 public interface Logger {
-    void log(String status);
+    void error(String error);
+    void fatal(String message);
+    void info(String message);
+    void trace(String message);
+    String retrieveLog();
```

Comment:
I'm not sure I'd consider it the responsibility of a logger to retrieve the logs. What if they were ephemeral, just printing to system out? Maybe your application logger could take store of some sort, and you could use that store elsewhere.